### PR TITLE
Refactored Find JVM to use functional programming.

### PR DESCRIPTION
**Pull request by DuarteMRAlves**

### CODE REVIEW

File: src/jvm.jl [hunk 1]
```diff
@@ -48,71 +29,129 @@ function findjvm()
...
+            l -> chain((pwd(),), l) |>
+            l -> flatmap(libpathfromdirectory, l)  |>
+            # Return only first path
+            first
+    catch err
+        errorMsg =
+            [
+            "Cannot find java library libjvm.dylib\n",
+            "Search Path:"
+            ];
```
Comment (tosmarcel) [inline src/jvm.jl]: MacOS specific extension in error message

File: src/jvm.jl [hunk 2]
```diff
@@ -48,71 +29,129 @@ function findjvm()
...
+        ]
+    end
 
+    function libpathfromdirectory(directory::AbstractString)::Vector{AbstractString}
+        libpath = joinpath(directory,"libjvm.dylib")
+        return (isfile(libpath) ? [libpath] : [])
+    end
 
+    function libpathfromdirectory(directory::AbstractString)::Vector{AbstractString}
+        libpath = joinpath(directory,"jvm.dll")
```
Comment (tosmarcel) [inline src/jvm.jl]: Leftover macos-specific definition

**Review by tosmarcel — CHANGES_REQUESTED** [review]

File: src/jvm.jl (hunk 1 above)
Comment (DuarteMRAlves) [reply src/jvm.jl]: Replaced file with variable set for each system

File: src/jvm.jl (hunk 2 above)
Comment (DuarteMRAlves) [reply src/jvm.jl]: Removed definition

File: src/jvm.jl [hunk 3]
```diff
@@ -48,71 +29,130 @@ function findjvm()
...
+            l -> flatmap(libpathdirectories, l) |> 
+            # Include pwd in directories to search for libpath
+            l -> chain((pwd(),), l) |>
+            l -> flatmap(libpathfromdirectory, l)  |>
+            # Return only first path
+            first
+    catch err
+        errormsg =
+            [
+            "Cannot find java library "*libfile*"\n",
```
Comment (tosmarcel) [inline src/jvm.jl]:
```suggestion
            "Cannot find java library $(libfile)\n",
```

**Review by tosmarcel — APPROVED** [review]

**Pull request merged**